src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -133,6 +136,10 @@ class AuthScreens extends React.Component {
     }
 
     shouldComponentUpdate(nextProps) {
+        if (this.props.currentPath !== nextProps.currentPath) {
+            App.setUpPoliciesAndNavigate(nextProps.session, nextProps.currentPath);
+        }
```

Comment:
Ah haha. My immediate thought was that it would be better to do this in `componentDidUpdate()` vs `shouldComponentUpdate()`. As latter is used to tell the component whether it should re-render and not really for side effects (though I guess we are doing similar stuff elsewhere :oh-nothing:).

The code on line 143 is preventing the AuthScreens from updating when the `currentPath` changes (which solves my concern shared in the original issue about unnecessary re-rendering). But also explains why we are not using `componentDidUpdate()` because it will only update if the `isSmallScreenWidth` props change. 😄 

Another option would be to allow the component to update when the `currentPath` changes and then we could use `componentDidUpdate()`, but that would trigger a re-render.

If we still prefer this logic to be here and not in the `NavigationRoot` I'd maybe add a comment or two to explain why we are doing this and why it needs to happen in `shouldComponentUpdate()`

Reply:
Yeah doing this here in `shouldComponentUpdate()` is a little micro-optimization. I'll add a comment to make that clear 👍🏽

Reply:
So to make sure that I understand this correctly, `App.setUpPoliciesAndNavigate` will be called once the NavigationRoot passes down the `currentPath`, which ensures that it only runs when the navigation is ready and the user is signed in. If that's correct then it looks good to me.

I think that would make this redundant, but it's probably wise to leave the check just in case.
https://github.com/Expensify/App/blob/0c2c2f64699f8845507f5e91506d5ca75d19dfee/src/libs/actions/App.js#L176-L178

Reply:
Actually wouldn't we be calling this any time the path changes? 🤔 and isn't that like every state change for the navigator? We just want it to run once though don't we?

Reply:
I just came across this code today while investigating https://github.com/Expensify/App/issues/10271 and I think it's a bad idea to put this logic inside of `shouldComponentUpdate()`. It's an anti-pattern (not what the purpose of the lifecycle method is for) and adds a side-effect in a place that's not really appropriate.

It also leads to this:

> Actually wouldn't we be calling this any time the path changes? 🤔 and isn't that like every state change for the navigator? We just want it to run once though don't we?

Which I think is a bad idea and not what is wanted.

Reply:
Do you mean this proposal? Not sure if you can see it or not. I can see it from the `conversation` page of the PR.

![image](https://user-images.githubusercontent.com/1228807/185479521-d51b0494-4094-43a9-acb8-ac0dea2323a3.png)

I was afraid someone was gonna ask me for a suggestion on how to fix it 😓 

Honestly, I'm a bit lost at everything that was done in this PR and I don't really understand how it fixes the original problem. I am going back now to read the GH closer as I see you struggled with understanding this in the beginning too. I'll see if I can follow along there.

Reply:
OK, I think I followed some of that. It looks like at some point in the conversation everyone agreed that `window.location.href` would work the best and be the most simple, but then, it was never attempted?

I think we should go that route. I agree with you that on native platforms, it can be a no-op and that's fine.

```
// index.js
function getCurrentUrl() {
    return window.location.href;
}

// index.native.js
function getCurrentUrl() {
    return '';
}
```

I wouldn't mind seeing how simple that would make all of this. Would it mean that all the changes in this PR can be reverted? I think that would be great if it's possible.

Reply:
Oh nevermind my old proposal is here in the issue :D https://github.com/Expensify/Expensify/issues/217505#issuecomment-1196040735

Reply:
@arosiclair as a matter of cleanup, could you look into switching this to `window.location.href` and see if that could simplify things and remove this side-effect?

Reply:
Yeah we can take the `window.location.href` approach since this can be messy. It's be a pretty simple change so I'll submit another PR and add you guys as reviewers